pkg/apis/upgrade.cattle.io/v1/types.go

```diff
@@ -60,12 +60,14 @@ type PlanStatus struct {
 
 // ContainerSpec is a simplified container template.
 type ContainerSpec struct {
-	Image   string                 `json:"image,omitempty"`
-	Command []string               `json:"command,omitempty"`
-	Args    []string               `json:"args,omitempty"`
-	Env     []corev1.EnvVar        `json:"envs,omitempty"`
-	EnvFrom []corev1.EnvFromSource `json:"envFrom,omitempty"`
-	Volumes []VolumeSpec           `json:"volumes,omitempty"`
+	Image           string                  `json:"image,omitempty"`
+	Command         []string                `json:"command,omitempty"`
+	Args            []string                `json:"args,omitempty"`
+	Env             []corev1.EnvVar         `json:"envs,omitempty"`
+	EnvFrom         []corev1.EnvFromSource  `json:"envFrom,omitempty"`
+	Volumes         []VolumeSpec            `json:"volumes,omitempty"`
+	SecurityContext *corev1.SecurityContext `json:"securityContext,omitempty"`
+	SELinuxOptions  *corev1.SELinuxOptions  `json:"seLinuxOptions,omitempty"`
```

Comment:
Just realized that SELinuxOptions is actually nested within the SecurityContext struct; we don't need a separate field for it. Is there any specific reason why you wanted to allow setting it separately? I guess I could see the value in allowing use of the default SecurityContext, but just overriding the SELinuxOptions. I could see this being confusing though, as it provides two places to set the SELinuxOptions - both in the SecurityContext's nested field, and in the top-level ContainerSpec field.